extensions/cornerstone/src/services/ViewportService/CornerstoneViewportService.ts

```diff
@@ -984,12 +988,21 @@ class CornerstoneViewportService extends PubSubService implements IViewportServi
           const backgroundDisplaySet = displaySetService.getDisplaySetsBy(
             displaySet =>
               !displaySet.isOverlayDisplaySet &&
-              displaySet.images.some(image => image.imageId === sampleImageId)
+              displaySet.images?.some(image => image.imageId === sampleImageId)
           );
 
           if (backgroundDisplaySet.length !== 1) {
             throw new Error('Background display set not found');
           }
+
+          if (viewport.type === csEnums.ViewportType.VOLUME_3D) {
+            debounceViewportCallback(() => {
```

Comment:
I first thought it is an actual debounce implementation, but i saw it is basically the setTimeout, can you rename it